src/mlpack/methods/gmm/diagonal_em_fit_impl.hpp

```diff
@@ -0,0 +1,387 @@
+/**
+ * @file diagonal_em_fit_impl.hpp
+ * @author Kim SangYeon
+ *
+ * Implementation of EM algorithm for fitting DiagonalGMMs.
```

Comment:
Can you point out that there are a couple of additional speedups over the usual EMFit here?  It took me a minute to compare the code and see what was different.  Additionally, it's up to you, but you could probably merge the `EMFit` and `DiagonalEMFit` class, using only the template type of the distribution to determine what computation to make for the update steps.

Reply:
Yeah, I'm going to merge the `EMFit` and `DiagonalEMFit` :)